Sources/BeeTracking/TrackingMetrics.swift

```diff
@@ -200,31 +213,45 @@ public struct TrackerEvaluationSequence {
 
 extension TrackerEvaluationSequence {
   /// Returns the performance of `tracker` on the sequence `self`.
-  public func evaluate(_ tracker: Tracker) -> SequenceEvaluationResults {
-    let subsequences = self.subsequences(deltaAnchor: 50)
-    let subsequenceEvaluations = subsequences.enumerated().map {
-      (i, subsequence) -> (metrics: SubsequenceMetrics, prediction: [OrientedBoundingBox]) in
-      print("Evaluating subsequence \(i + 1) of \(subsequences.count)")
-      return subsequence.evaluateSubsequence(tracker)
-    }
-    return SequenceEvaluationResults(
-      subsequences: subsequenceEvaluations,
-      sequenceMetrics: SequenceMetrics(subsequenceEvaluations.map { $0.metrics }))
-  }
-
-  /// Returns the performance of `tracker` on the subsequence `self`.
-  public func evaluateSubsequence(_ tracker: Tracker)
-    -> (metrics: SubsequenceMetrics, prediction: [OrientedBoundingBox])
-  {
+  public func evaluate(_ tracker: Tracker, deltaAnchor: Int, outputFile: String) -> SequenceEvaluationResults {
     guard let _ = try? Python.attemptImport("shapely") else {
       print("python shapely library must be installed")
       preconditionFailure()
     }
 
-    let prediction = tracker(frames, groundTruth[0])
-    return (
-      metrics: SubsequenceMetrics(groundTruth: groundTruth, prediction: prediction),
-      prediction: prediction)
+    let subsequences = self.subsequences(deltaAnchor: deltaAnchor)
+    let subsequencePredictions = [[OrientedBoundingBox]](
+      unsafeUninitializedCapacity: subsequences.count
+    ) { (buf, actualCount) in
+      let blockCount = 4
+      ComputeThreadPools.local.parallelFor(n: blockCount) { (blockIndex, _) in
+        for i in 0..<subsequences.count {
+          guard i % (2 * blockCount) == blockIndex
+            || i % (2 * blockCount) == 2 * blockCount - 1 - blockIndex
+          else {
+            continue
+          }
+          let subsequence = subsequences[i]
+          print("Evaluating subsequence \(i + 1) of \(subsequences.count)")
+          (buf.baseAddress! + i).initialize(to: tracker(subsequence.frames, subsequence.groundTruth[0]))
```

Comment:
This bit is a bit hacky and unexplained. It tries to parallelize mutliple subsequence evaluations using penguin parallel, while limiting the parallelism to 4. I think we should eventually simplify it by adding a feature to penguin that supports limiting the parallelism.

I'll create a PR with a comment and a TODO.